lib/babe/babe.go

```diff
@@ -478,19 +478,13 @@ func (b *Service) getParentForBlockAuthoring(slotNum uint64) (*types.Header, err
 	return parent, nil
 }
 
-func (b *Service) handleSlot(epoch, slotNum uint64,
+func (b *Service) handleSlot(epoch uint64, slot Slot,
 	authorityIndex uint32,
 	preRuntimeDigest *types.PreRuntimeDigest,
 ) error {
-	currentSlot := Slot{
-		start:    time.Now(),
-		duration: b.constants.slotDuration,
-		number:   slotNum,
-	}
-
-	parent, err := b.getParentForBlockAuthoring(slotNum)
+	parent, err := b.getParentForBlockAuthoring(slot.number)
 	if err != nil {
-		return fmt.Errorf("could not get parent for claiming slot %d: %w", slotNum, err)
+		return fmt.Errorf("could not get parent for claiming slot %d: %w", slot.number, err)
```

Comment:
nit whilst we're at it:

```suggestion
		return fmt.Errorf("getting parent for block authoring for claiming slot %d: %w", slot.number, err)
```